src/zkp/pimod.rs

```diff
@@ -6,7 +6,30 @@
 // License, Version 2.0 found in the LICENSE-APACHE file in the root directory
 // of this source tree.
 
-//! Implements the ZKP from Figure 16 of <https://eprint.iacr.org/2021/060.pdf>
+//! Implements a zero-knowledge proof that the modulus N is a Paillier-Blum
+//! modulus meaning that the gcd (N, phi(N)) = 1 where phi is the Euler's
+//! totient function and `N = pq` where `p`, `q` is congruent to `3 mod 4`.
+//!
+//! The protocol is a combination (and simplification) of van de Graaf and
+//! Peralta[^cite1] and Goldberg et al[^cite2].
+//!
+//! The proof is defined in Figure 16 of CGGMP[^cite].
+//!
+//! This implementation uses a standard Fiat-Shamir transformation to make the
+//! proof non-interactive.
+//!
+//! [^cite]: Ran Canetti, Rosario Gennaro, Steven Goldfeder, Nikolaos Makriyannis, and Udi Peled.
+//! UC Non-Interactive, Proactive, Threshold ECDSA with Identifiable Aborts.
+//! [EPrint archive, 2021](https://eprint.iacr.org/2021/060.pdf).
+//! [^cite1]: J. van de Graaf and R. Peralta. A simple and secure way to show the validity of your public key. In
+//! Advances in Cryptology - CRYPTO ’87, A Conference on the Theory and
+//! Applications of Cryptographic Techniques, Santa Barbara, California, USA,
+//! August 16-20, 1987, Proceedings, pages 128–134, 1987. [^cite2]: Sharon
+//! Goldberg, Leonid Reyzin, Omar Sagga, and Foteini Baldimtsi. Efficient
+//! noninteractive certification of RSA moduli and beyond. In Advances in
+//! Cryptology - ASIACRYPT 2019 - 25th International Conference on the Theory
+//! and Application of Cryptology and Information Security, Kobe, Japan,
+//! December 8-12, 2019, Proceedings, Part III, pages 700–727, 2019.
 
```

Comment:
todo: These footnotes aren't building quite right. You can see if you build the docs
```
$ cargo doc --release --no-deps --workspace --with-private-items --open
```

See the [footnotes section](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#footnotes) of rustdoc documentation.

Reply:
This is still not building correctly. You need newlines between each citation down here
```rust
//! [^cite]: Ran Canetti,...
//!
//! [^cite1]: J. van de Graaf...
//!
//! [^cite2]: Sharon Goldberg...
```